Src/library/net/network/socket/network_socket.e

```diff
@@ -163,16 +163,33 @@ feature -- Status report
 		local
 			retval: INTEGER
 		do
-			retval := c_check_exception_with_timeout (descriptor, timeout)
+			retval := c_check_exception_with_timeout (descriptor, timeout_in_nano_seconds)
 			Result := (retval > 0)
 		end
 
 	timeout: INTEGER
 			-- Duration of timeout in seconds
 
+	timeout_additional_nano_seconds: INTEGER
+			-- Additional nanoseconds to `timeout`.
+			-- between 0 and < 1_000_000_000 nanoseconds = 1 second.
+
+	timeout_in_nano_seconds: NATURAL_64
```

Comment:
I would have done the opposite. I would have made `timeout` an obsolete function returning the new attribute `timeout_ns` (shorter but still clear name instead of `timeout_in_nano_seconds`).